.github/workflows/ci.yml

```
@@ -4,16 +4,39 @@ name: CI
+      - "**.md"
+      - "docs/**"
+      - ".kiro/**"
 
 jobs:
+  # Fast syntax check - fails early to save costs
+  syntax-check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Python 3.9
+        uses: actions/setup-python@v4
```

actionlint (1.7.7)

the runner of "actions/setup-python@v4" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)